shell/packages/sandstorm-db/db.js

```diff
@@ -1153,4 +1179,74 @@ if (Meteor.isServer) {
       }
     }
   };
+
+  var ValidKeyFingerprint = Match.Where(function (keyFingerprint) {
+    check(keyFingerprint, String);
+    return !!keyFingerprint.match(/^[0-9A-F]{40}$/);
+  });
+
+  SandstormDb.prototype.updateKeybaseProfileAsync = function (keyFingerprint) {
+    // Asynchronously fetch the given Keybase profile and populate the KeybaseProfiles collection.
+
+    check(keyFingerprint, ValidKeyFingerprint);
+
+    console.log("fetching keybase", keyFingerprint);
+
+    HTTP.get(
+        "https://keybase.io/_/api/1.0/user/lookup.json?key_fingerprint=" + keyFingerprint +
+        "&fields=basics,profile,proofs_summary", {
+      timeout: 5000
+    }, function (err, keybaseResponse) {
+      if (err) {
+        console.log("keybase lookup error:", err.stack);
+        return;
+      }
+
+      if (!keybaseResponse.data) {
+        console.log("keybase didn't return JSON? Headers:", keybaseResponse.headers);
+        return;
+      }
+
+      var profile = (keybaseResponse.data.them || [])[0];
+
+      if (profile) {
+        var record = {
+          displayName: (profile.profile || {}).full_name,
+          handle: (profile.basics || {}).username,
+          proofs: (profile.proofs_summary || {}).all || []
+        };
+
+        record.proofs.forEach(function (proof) {
+          // Remove potentially Mongo-incompatible stuff. (Currently Keybase returns nothing that
+          // this would filter.)
+          for (var field in proof) {
+            if (field.match(/[.$]/) || typeof(proof[field]) === "object") {
+              delete proof[field];
+            }
+          }
+
+          // Indicate not verified.
+          // TODO(security): Asynchronously verify proofs. Presumably we can borrow code from the
+          //   Keybase node-based CLI.
+          proof.status = "unverified";
+        });
+
+        KeybaseProfiles.update(keyFingerprint, {$set: record}, {upsert: true});
+      } else {
+        KeybaseProfiles.update(keyFingerprint, {$unset: {profile: ""}}, {upsert: true});
```

Comment:
Not sure where this `profile` field that you're unsetting is coming from - it's not declared in `db.js`.  Perhaps this should be `displayName`?